release/models/platform/openconfig-platform-healthz-fault.yang

```diff
@@ -0,0 +1,311 @@
+  }
+
+  identity ACTION_COLD_REBOOT {
+    base ACTION_BASE;
+    description
+      "Reboot that interrupts device operation, ie a gNOI.System.RebootRequest
+      with COLD RebootMethod";
+  }
+
+  identity ACTION_POWER_CYCLE {
+    base ACTION_BASE;
+    description
+      "The process of turning a component off completely and then turning it
+      back on.
+
+      This could be acheived with two gNOI.System.RebootRequest calls:
+      a POWERDOWN RebootMethod followed by a POWERUP RebootMethod.
+
+      At the device level, using a remote controlled power strip to
+      mechanically disconnect the power source (ie: pull the plug) is another
+      example";
+  }
+
+  identity ACTION_FACTORY_RESET {
+    base ACTION_BASE;
+    description
+      "Restores a component to its original factory settings.
+      At the device level, this could be acheived with the gNOI.FactoryReset
+      service";
+  }
+
+  identity ACTION_REPLACE {
+    base ACTION_BASE;
+    description
+      "Replace component.";
+  }
+
+  grouping platform-health-faults {
+    description
+      "Grouping containing a component's active faults";
+
+    container faults {
+      description
+        "Enclosing container for a component's active faults.";
+
+      list fault {
+        key "symptom";
+        config false;
+        description
+          "A fault indicates that a component has recently or is currently
+          experiencing a problem as expressed by a symptom. These symptoms have
+          underlying conditions that are platform-specific.
+
+          Faults should only be raised if the component is no longer able to
+          perform the function expected of it and requires
+          remediation/intervention.
+
+          The fault should not be deleted as soon as the underlying condition
+          is no longer asserted to avoid excessive fault creation and deletion.
+          Deletion of the fault should occur after a suitable period outside of
+          the triggering condition";
+
+        leaf symptom {
+          config false;
+          description
+           "The symptom that raised this fault";
+          type leafref {
+            path "../state/symptom";
+          }
+        }
+
+        container state {
+          config false;
+          description
+            "Operational fault state.";
+          uses fault-state;
+        }
+
+        uses fault-remediations;
+      }
+    }
+  }
+
+  grouping fault-state {
+    description
+      "The fault's operational state";
+    leaf symptom {
+      type identityref {
+        base SYMPTOM_BASE;
+      }
+      description
+        "The symptom that raised this fault";
+    }
+
+    leaf origin-time {
+      type oc-types:timeticks64;
+      description
+        "The time at which the fault first became active.
+        This value is static for the life of the fault.";
+    }
+
+    leaf last-detection-time {
+      type oc-types:timeticks64;
+      description
```

Comment:
Per `oc-types:timeticks64` definition, this leaf must define the epoch referenced (Can see other uses for examples)